src/main/java/org/apache/mesos/offer/MesosResourcePool.java

```diff
@@ -63,14 +86,19 @@ public MesosResource consume(ResourceRequirement resReq) {
 
         logger.error("The following resource requirement did not meet any consumption criteria: {}",
                 TextFormat.shortDebugString(resReq.getResource()));
-        return null;
+        return Optional.empty();
     }
 
-    public MesosResource consume(DynamicPortRequirement dynamicPortRequirement) {
+    /**
+     * Consumes and returns a {@link MesosResource} which meets the provided
+     * {@link DynamicPortRequirement}, or does nothing and returns an empty {@link Optional} if no
+     * available resources meet the requirement.
+     */
+    public Optional<MesosResource> consume(DynamicPortRequirement dynamicPortRequirement) {
         Value availableValue = unreservedMergedPool.get(dynamicPortRequirement.getName());
 
         if (availableValue == null) {
-            return null;
+            return Optional.empty();
```

Comment:
I'd leaned towards doing it for the public consume() methods as they make the intended behavior a little clearer.

Ended up carrying it over to the private methods where relevant since their return values were effectively passed up through the consume() methods.